### ISSUE TITLE
Bug? Command Parse Failed: Element: (0000,0700) Priority

### ISSUE THREAD

**Issue by MichaelLeeHobbs** [PantelisGeorgiadis/dcmjs-dimse#9]
It seems like the "Command Field" "Priority" is not being set. References: 
https://dicom.nema.org/dicom/2013/output/chtml/part07/chapter_E.html
https://dicom.nema.org/dicom/2013/output/chtml/part07/chapter_9.html

It seems like it's a required item but not sure.

dcmjs-dimse
```
const dcmjsDimse = require('dcmjs-dimse');
const { Client } = dcmjsDimse;
const { CStoreRequest } = dcmjsDimse.requests;

const client = new Client();
const request = new CStoreRequest('0091-21122416.dcm');
client.addRequest(request);
client.on('networkError', (e) => {
  console.log('Network error: ', e);
});
client.send('127.0.0.1', 104, 'SCU', 'ANY-SCP');
```

Store SCP cmd `storescp.exe 104 --log-level trace --accept-all --promiscuous`

Error: `E: DIMSE failure (aborting association): 0006:020b DIMSE: Command Parse Failed: Element: (0000,0700) Priority`

Full debug
```
Start: C:\ProgramData\chocolatey\bin\storescp.exe 104 --log-level trace --accept-all --promiscuous

D: $dcmtk: storescp v3.6.6 2021-01-14 $
stdout data: T:   environment variable TCP_BUFFER_LENGTH not set, using the system defaults

T: checking whether environment variable TCP_NODELAY is set
T:   environment variable TCP_NODELAY not set, using the default value (0)

stdout data: D: Association Received: db
D: setting network send timeout to 60 seconds
D: setting network receive timeout to 60 seconds
T: DUL  FSM Table: State: 1 Event: 4
T: DUL  Event:  Transport connection indication
T: DUL  Action: AE 5 Transport Connect Response
T: setting timeout for first PDU to be read to 30 seconds

stdout data: T: Read PDU HEAD TCP: 01 00 00 00 01 2b
T: Read PDU HEAD TCP: type: 01, length: 299 (12b)
T: DUL  FSM Table: State: 2 Event: 5
T: DUL  Event:  A-ASSOCIATE-RQ PDU (on transport)
T: DUL  Action: AE 6 Examine Associate Request
D: PDU Type: Associate Request, PDU Length: 299 + 6 bytes PDU header
D:   01  00  00  00  01  2b  00  01  00  00  41  4e  59  2d  53  43
D:   50  20  20  20  20  20  20  20  20  20  53  43  55  20  20  20
D:   20  20  20  20  20  20  20  20  20  20  00  00  00  00  00  00
D:   00  00  00  00  00  00  00  00  00  00  00  00  00  00  00  00
D:   00  00  00  00  00  00  00  00  00  00  10  00  00  15  31  2e
D
stdout data: :   32  2e  38  34  30  2e  31  30  30  30  38  2e  33  2e  31  2e
D:   31  2e  31  20  00  00  4d  01  00  00  00  30  00  00  19  31
D:   2e  32  2e  38  34  30  2e  31  30  30  30  38  2e  35  2e  31
D:   2e  34  2e  31  2e  31  2e  31  40  00  00  11  31  2e  32  2e
D:   38  34  30  2e  31  30  30  30  38  2e  31  2e  32  40  00  00
D:   13  31  2e  32  2e  38  34  30  2e  31  30  30  30  38  2e  31
D:   2e  32  2e  31  20  00  00  3b  03  00  00  00  30  00  00  19
D:   31  2e  32  2e  38  34  30  2e  31  30  30  30  38  2e  35  2e
D:   31  2e  34  2e  31  2e  31  2e  31  40  00  00  16  31  2e  32
D:   2e  38  34  30  2e  31  30  30  30  38  2e  31  2e  32  2e  34
D:   2e  39  30  50  00  00  3a  51  00  00  04  00  04  00  00  52
D:   00  00  1a  31  2e  32  2e  38  32  36  2e  30  2e  31  2e  33
D:   36  38  30  30  34  33  2e  31  30  2e  38  35  34  55  00  00
D:   10  44  43  4d  4a  53  2d  44  49  4d  53  45  2d  56  30  2e
D:   31
D: Parsing an A-ASSOCIATE PDU
T: PDU type: 1 (A-ASSOCIATE RQ), PDU Length: 299
T: DICOM Protocol: 1
T: Called AP Title:  ANY-SCP
T: Calling AP Title: SCU
T: Parsing remaining 231 bytes of A-ASSOCIATE PDU
T: Next item type: 10
T: Subitem parse: Type 10, Length 0021, Content: 1.2.840.10008.3.1.1.1
T: Successfully parsed Application Context
T: Parsing remaining 206 bytes of A-ASSOCIATE PDU
T: Next item type: 20
T: Parsing Presentation Context: (20), Length: 77
T: Presentation Context ID: 01
T: Parsing remaining 73 bytes of Presentation Context
T: Next item type: 30
T: Subitem parse: Type 30, Length 0025, Content: 1.2.840.10008.5.1.4.1.1.1
T: Successfully parsed Abstract Syntax
T: Parsing remaining 44 bytes of Presentation Context
T: Next item type: 40
T: Subitem parse: Type 40, Length 0017, Content: 1.2.840.10008.1.2
T: Successfully parsed Transfer Syntax
T: Parsing remaining 23 bytes of Presentation Context
T: Next item type: 40
T: Subitem parse: Type 40, Length 0019, Content: 1.2.840.10008.1.2.1
T: Successfully parsed Transfer Syntax
T: Successfully parsed Presentation Context
T: Parsing remaining 125 bytes of A-ASSOCIATE PDU
T: Next item type: 20
T: Parsing Presentation Context: (20), Length: 59
T: Presentation Context ID: 03
T: Parsing remaining 55 bytes of Presentation Context
T: Next item type: 30
T: Subitem parse: Type 30, Length 0025, Content: 1.2.840.10008.5.1.4.1.1.1
T: Successfully parsed Abstract Syntax
T: Parsing remaining 26 bytes of Presentation Context
T: Next item type: 40
T: Subitem parse: Type 40, Length 0022, Content: 1.2.840.10008.1.2.4.90
T: Successfully parsed Transfer Syntax
T: Successfully parsed Presentation Context
T: Parsing remaining 62 bytes of A-ASSOCIATE PDU
T: Next item type: 50
T: Parsing user info field (50), Length: 58
T: Parsing remaining 58 bytes of User Information
T: Next item type: 51
T: Maximum PDU Length: 262144
T: Successfully parsed Maximum PDU Length
T: Parsing remaining 50 bytes of User Information
T: Next item type: 52
T: Subitem parse: Type 52, Length 0026, Content: 1.2.826.0.1.3680043.10.854
T: Parsing remaining 20 bytes of User Information
T: Next item type: 55
T: Subitem parse: Type 55, Length 0016, Content: DCMJS-DIMSE-V0.1
T: Successfully parsed User Information
I: Association Received
D: Parameters:
D: ====================== BEGIN A-ASSOCIATE-RQ =====================
D: Our Implementation Class UID:      1.2.276.0.7230010.3.0.3.6.6
D: Our Implementation Version Name:   OFFIS_DCMTK_366
D: Their Implementation Class UID:    1.2.826.0.1.3680043.10.854
D: Their Implementation Version Name: DCMJS-DIMSE-V0.1
D: Application Context Name:    1.2.840.10008.3.1.1.1
D: Calling Application Name:    SCU
D: Called Application Name:     ANY-SCP
D: Responding Application Name:
D: Our Max PDU Receive Size:    16384
D: Their Max PDU Receive Size:  262144
D: Presentation Contexts:
D:   Context ID:        1 (Proposed)
D:     Abstract Syntax: =ComputedRadiographyImageStorage
D:     Proposed SCP/SCU Role: Default
D:     Proposed Transfer Syntax(es):
D:       =LittleEndianImplicit
D:       =LittleEndianExplicit
D:   Context ID:        3 (Proposed)
D:     Abstract Syntax: =ComputedRadiographyImageStorage
D:     Proposed SCP/SCU Role: Default
D:     Proposed Transfer Syntax(es):
D:       =JPEG2000LosslessOnly
D: Requested Extended Negotiation: none
D: Accepted Extended Negotiation:  none
D: Requested User Identity Negotiation: none
D: User Identity Negotiation Response:  none
D: ======================= END A-ASSOCIATE-RQ ======================
T: DUL  FSM Table: State: 3 Event: 6
T: DUL  Event:  A-ASSOCIATE resp prim (accept)
T: DUL  Action: AE 7 Send Associate AC
D: Constructing Associate AC PDU
I: Association Acknowledged (Max Send PDV: 131060)
D: ====================== BEGIN A-ASSOCIATE-AC =====================
D: Our Implementation Class UID:      1.2.276.0.7230010.3.0.3.6.6
D: Our Implementation Version Name:   OFFIS_DCMTK_366
D: Their Implementation Class UID:    1.2.826.0.1.3680043.10.854
D: Their Implementation Version Name: DCMJS-DIMSE-V0.1
D: Application Context Name:    1.2.840.10008.3.1.1.1
D: Calling Application Name:    SCU
D: Called Application Name:     ANY-SCP
D: Responding Application Name: ANY-SCP
D: Our Max PDU Receive Size:    16384
D: Their Max PDU Receive Size:  262144
D: Presentation Contexts:
D:   Context ID:        1 (Accepted)
D:     Abstract Syntax: =ComputedRadiographyImageStorage
D:     Proposed SCP/SCU Role: Default
D:     Accepted SCP/SCU Role: Default
D:     Accepted Transfer Syntax: =LittleEndianExplicit
D:   Context ID:        3 (Accepted)
D:     Abstract Syntax: =ComputedRadiographyImageStorage
D:     Proposed SCP/SCU Role: Default
D:     Accepted SCP/SCU Role: Default
D:     Accepted Transfer Syntax: =JPEG2000LosslessOnly
D: Requested Extended Negotiation: none
D: Accepted Extended Negotiation:  none
D: Requested User Identity Negotiation: none
D: User Identity Negotiation Response:  none
D: ======================= END A-ASSOCIATE-AC ======================
T: DIMSE receiveCommand

stdout data: T: Read PDU HEAD TCP: 04 00 00 00 00 8c
T: Read PDU HEAD TCP: type: 04, length: 140 (8c)
T: DUL  FSM Table: State: 6 Event: 9
T: DUL  Event:  P-DATA-TF PDU (on transport)
T: DUL  Action: DT 2 Indicate P DATA PDU Received
D: DcmDataset::read() TransferSyntax="Little Endian Implicit"
T: DcmItem::insert() Element (0000,0000) VR="UL" inserted at beginning
T: DcmItem::readSubItem() returns error = Normal
T: DcmItem::insert() Element (0000,0002) VR="UI" inserted
T: DcmItem::readSubItem() returns error = Normal
T: DcmItem::insert() Element (0000,0100) VR="US" inserted
T: DcmItem::readSubItem() returns error = Normal
T: DcmItem::insert() Element (0000,0110) VR="US" inserted
T: DcmItem::readSubItem() returns error = Normal
T: DcmItem::insert() Element (0000,0800) VR="US" inserted
T: DcmItem::readSubItem() returns error = Normal
T: DcmItem::insert() Element (0000,1000) VR="UI" inserted
T: DcmItem::readSubItem() returns error = Normal
T: DcmItem::read() returns error = Normal
T: DcmDataset::read() returns error = Normal
T: DIMSE receiveCommand: 1 PDVs (134 bytes), PresID=3
T: DIMSE Command Received:
T:
T: # Dicom-Data-Set
T: # Used TransferSyntax: Little Endian Implicit
T: (0000,0002) UI =ComputedRadiographyImageStorage         #  26, 1 AffectedSOPClassUID
T: (0000,0100) US 1                                        #   2, 1 CommandField
T: (0000,0110) US 1                                        #   2, 1 MessageID

stdout data: T: (0000,0800) US 514                                      #   2, 1 CommandDataSetType
T: (0000,1000) UI [1.3.12.2.1107.5.3.49.22789.11.202111291127350437.1] #  50, 1 AffectedSOPInstanceUID
T:
T: DcmItem::searchSubFromHere() Element (0000,0100) found
T: DcmItem::searchSubFromHere() Element (0000,0100) found
T: DcmItem::searchSubFromHere() Element (0000,0110) found
T: DcmItem::searchSubFromHere() Element (0000,0800) found
T: DcmItem::searchSubFromHere() Element (0000,0002) found
T: DcmItem::searchSubFromHere() Element (0000,1000) found
E: DIMSE failure (aborting association): 0006:020b DIMSE: Command Parse Failed: Element: (0000,0700) Priority
T: DUL  FSM Table: State: 6 Event: 14
T: DUL  Event:  A-ABORT request primitive
T: DUL  Action: AA 1 Send A ABORT PDU

stdout data: T: Read PDU HEAD TCP: 04 00 00 00 40 00
T: Read PDU HEAD TCP: type: 04, length: 16384 (4000)
T: DUL  FSM Table: State: 13 Event: 9
T: DUL  Event:  P-DATA-TF PDU (on transport)
T: DUL  Action: AA 6 Ignore PDU
```

**Comment by PantelisGeorgiadis** [comment]
Thank you @MichaelLeeHobbs for your detailed message! You are absolutely right! Based on my research, the Command Message Field "Priority" is legal just for DIMSE-C Requests except C-ECHO and not for DIMSE-N Requests. The library was updated accordingly (v0.0.12).

**Comment by MichaelLeeHobbs** [comment]
Confirmed fixed in 0.0.12. Excellent work btw! Been looking for something like this for years. DCMTK storescp is difficult to use and dcmsend is a bit too limited. Will keep an eye on and start testing dcmjs-dimse.

**Comment by PantelisGeorgiadis** [comment]
Thank you Sir!
